### PR TITLE
Corrections to errors in the existing files

### PR DESCRIPTION
**Pull request by kvedala**
#### Description of Change

There are indeed a lot of corrections because of the errors that already exist in the repository. 
This is a major effort to fix all the existing errors in the repository:
* wrongfully named
* non-compiling codes
* cpplint incompatible codes

There are a lot of files that changed and I already have a few auto-check tools that verify the CPPLINT and also compilation on linux, Windows and Mac operating systems. This can be checked here: https://github.com/kvedala/C-Plus-Plus/tree/cmake

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by kvedala** [comment]
@cclauss  seems like the cpplint check tool is incorrectly linking the files: [link](https://github.com/TheAlgorithms/C-Plus-Plus/runs/708030604#step:9:106)

> subprocess.CalledProcessError: Command '['g++', 'computer_oriented_statistical_methods/Gaussian_elimination.cpp', 'computer_oriented_statistical_methods/bisection_method.cpp', 'computer_oriented_statistical_methods/false-position.cpp', 'computer_oriented_statistical_methods/newton_raphson_method.cpp', 'computer_oriented_statistical_methods/secant_method.cpp', 'computer_oriented_statistical_methods/successive_approximation.cpp', 'math/fibonacci.cpp', 'math/prime_factorization.cpp', 'others/GCD_of_n_numbers.cpp', 'others/fibonacci_fast.cpp', 'others/matrix_exponentiation.cpp', 'others/pascal_triangle.cpp', 'sorting/shell_sort2.cpp']' returned non-zero exit status 1.
```

File: .github/workflows/cpplint_modified_files.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
   cpplint_modified_files:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1  # v2 is broken for git diff
+      - uses: actions/checkout@v1 # v2 is broken for git diff
```
Comment (cclauss) [inline .github/workflows/cpplint_modified_files.yml:14]: Please revert.  Two spaces is clearer to the reader.

File: .github/workflows/cpplint_modified_files.yml [hunk 2]
```diff
@@ -42,7 +42,8 @@ jobs:
           print("g++:")
           # compile_exts = tuple(".c .c++ .cc .cpp .cu .cxx".split())
           # compile_files = [file for file in cpp_files if file.lower().endswith(compile_exts)]
-          subprocess.run(["g++"] + cpp_files, check=True, text=True)
+          for cpp_file in cpp_files:
+            subprocess.run(["g++", cpp_file], check=True, text=True)
```
Comment (cclauss) [inline .github/workflows/cpplint_modified_files.yml:46]: Why is this deceleration of the script useful?

File: .gitignore [hunk 3]
```diff
@@ -33,3 +33,5 @@
 a.out
 *.out
 *.app
+
+build/
```
Comment (cclauss) [inline .gitignore:37]: The 🚫 means the the file does not end with a \n

**Comment by cclauss** [comment]
Too many files to review on one pull request.  Can someone else please review?

File: .github/workflows/cpplint_modified_files.yml (hunk 2 above)
Comment (kvedala) [reply .github/workflows/cpplint_modified_files.yml:46]: This fixes the above error wherein g++ is compiling multiple files simultaneously

File: .github/workflows/cpplint_modified_files.yml (hunk 1 above)
Comment (kvedala) [reply .github/workflows/cpplint_modified_files.yml:14]: This is per the clang-format specs. But if this is a one-off change, will change it

File: .github/workflows/cpplint_modified_files.yml (hunk 2 above)
Comment (kvedala) [reply .github/workflows/cpplint_modified_files.yml:46]: Link to the error details: https://github.com/TheAlgorithms/C-Plus-Plus/pull/795#issuecomment-633790790

File: .gitignore (hunk 3 above)
Comment (kvedala) [reply .gitignore:37]: Agreed, will fix

File: .github/workflows/cpplint_modified_files.yml (hunk 1 above)
Comment (cclauss) [reply .github/workflows/cpplint_modified_files.yml:14]: The is .yml, not .cpp ;-)

File: .github/workflows/cpplint_modified_files.yml (hunk 2 above)
Comment (cclauss) [reply .github/workflows/cpplint_modified_files.yml:46]: Makes sense now.

**Comment by kvedala** [comment]
> Too many files to review on one pull request. Can someone else please review?

There are a lot of errors to be fixed :sweat:

**Comment by kvedala** [comment]
included in #805

**Pull request closed without merging**